File: CodalityTestHarness/CodifyTestHarness/5. Prefix Sums/PassingCars.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace CodalityTestLibrary
{
    /// <summary>
    /// Write a function that, given a non-empty zero-indexed array A of N 
    /// integers, returns the number of pairs of passing cars
    /// </summary>
    public class PassingCars
    {
        public int solution(int[] numericArray)
        {
            int passCount = 0;

            for (int i = 0; i < numericArray.Length; i++)
            {
                if (numericArray[i] == 0)
                {
                    passCount += numericArray.Skip(i).Sum();
                }
            }

            return passCount;
        }
    }
}

File: Python/5. Prefix Sums/PassingCars.py
# Write a function that, given a non-empty zero-indexed array A of N 
# integers, returns the number of pairs of passing cars

class PassingCars(object):

    def solution(self, numericArray):

        passCount = 0

        for index, value in enumerate(numericArray):
            if value == 0:
                passCount += sum(numericArray[index:])
        
        return passCount